### ISSUE TITLE
q-date navigation-min-year-month error

### ISSUE THREAD

**Issue by HyeongRae** [quasarframework/quasar#12032]
### What happened?

q-date components make error if set default-view='Months', emit-immediately='true' and navigation-min-year-month '2019/12' or '2021/8' or '2020/11' or etc.
In these months, the first day is on a Sunday.

### What did you expect to happen?

not make error.

### Reproduction URL

https://codepen.io/hyeongrae/pen/poWqLaN

### How to reproduce?

1. click date
2. click '<' btn until 2019 year
3. click  DEC
4. find error

### Flavour

Quasar CLI (@quasar/cli | @quasar/app)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome

### Quasar info output

```shell
Operating System - Darwin(21.1.0) - darwin/x64
NodeJs - 16.13.1

Global packages
  NPM - 8.1.2
  yarn - 1.22.17
  @quasar/cli - 1.1.3
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.4.3 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 3.2.6 -- Quasar Framework local CLI
  @quasar/extras - 1.12.2 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.26 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.0.12
  vuex - 4.0.2 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.16.7 -- Babel compiler core.
  webpack - 5.65.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.7.1 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - 6.4.2 -- A plugin for your Webpack build process, helping you generate a manifest of local files that workbox-sw should precache.
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.4 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - gimhyeonglaeui-MacBookPro.local
  en0 - 192.168.1.60
```

### Relevant log output

```shell
QDate.js?52ee:564 Uncaught (in promise) TypeError: Cannot convert undefined or null to object
    at Function.assign (<anonymous>)
    at eval (QDate.js?52ee:564:1)
    at Array.forEach (<anonymous>)
    at ReactiveEffect.eval [as fn] (QDate.js?52ee:562:1)
    at ReactiveEffect.run (reactivity.esm-bundler.js?0a2b:160:1)
    at ComputedRefImpl.get value [as value] (reactivity.esm-bundler.js?0a2b:1090:1)
    at eval (QDate.js?52ee:1211:1)
    at eval (runtime-core.esm-bundler.js?9e79:2873:1)
    at renderFnWithContext (runtime-core.esm-bundler.js?9e79:421:1)
    at Proxy.eval (runtime-core.esm-bundler.js?9e79:2873:1)
```

**Comment by pdanpdan** [comment]
Please reduce this to a minimal reproduction case

**Comment by HyeongRae** [comment]
> Please reduce this to a minimal reproduction case

https://codepen.io/hyeongrae/pen/poWqLaN
i modified the code

**Comment by rstoenescu** [comment]
It's erroring out because you're not using it as it's meant to be used. Meaning with a full date (year + month + day).
But regardless, fixed this case in Quasar v2.4.10 (to be released).
